pkg/controllers/inventory_controller.go

```diff
@@ -425,3 +364,33 @@ func (r *InventoryReconciler) housekeepingBMCJob(ctx context.Context, i *seederv
 
 	return nil
 }
+
+func (r *InventoryReconciler) triggerPowerAction(ctx context.Context, iObj *seederv1alpha1.Inventory) error {
+	i := iObj.DeepCopy()
+	if i.Status.Status == seederv1alpha1.InventoryReady && !util.ConditionExists(i.Status.Conditions, seederv1alpha1.BMCJobSubmitted) && i.Spec.PowerActionRequested != "" && i.Status.PowerAction.LastJobName == "" {
```

Comment:
Here we check the `.status.powerAction.lastJobName`'s value to decide whether it should handle the requested power action for an Inventory CR. We do set the value for it, but I can't find a place where it empties the value. From my testing result, I can only do power action once (by adding a `.spec.powerActionRequested: shutdown`), and then the subsequent requests, e.g. `.spec.powerActionRequested: poweron`, are not handled by the controller.

Maybe I used it wrong. Would you explain this a bit? Thanks!

Reply:
the reset is done from the harvester power action API: https://github.com/harvester/harvester/pull/3738/files#diff-eb153a88748df1fbf46481b843d26f5882036ceca1c94824bf9fecfcd649c862R339-R347